File: algorithm/src/main/java/baekjoon/p01100/Main.java
package baekjoon.p01100;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        char[][] inputs = new char[8][8];

        for (int i = 0; i < inputs.length; i++) {
            String input = sc.next();
            for (int j = 0; j < input.length(); j++) {
                inputs[i][j] = input.charAt(j);
            }
        }

        System.out.println(solve(inputs));
    }

    private static int solve(char[][] inputs) {
        int ret = 0;

        for (int i = 0; i < inputs.length; i++) {
            for (int j = 0; j < inputs[i].length; j++) {
                if (inputs[i][j] == 'F' && isWhite(i, j)) {
                    ret++;
                }
            }
        }

        return ret;
    }

    private static boolean isWhite(int i, int j) {
        if (i % 2 == 0) {
            return (j % 2 == 0);
        } else {
            return (j % 2 == 1);
        }
    }

}

File: algorithm/src/main/java/baekjoon/p01100/Main.kt
package baekjoon.p01100

fun main(args: Array<String>) {
    val inputs = List(8) { readLine()!!.toCharArray() }
    println(solve(inputs))
}

private fun solve(inputs: List<CharArray>): Int {
    var ret = 0

    for (i in 0 until inputs.size) {
        for (j in 0 until inputs[i].size) {
            if (inputs[i][j] == 'F' && isWhite(i, j)) {
                ret++
            }
        }
    }

    return ret
}

private fun isWhite(i: Int, j: Int): Boolean {
    return when {
        (i % 2 == 0) -> j % 2 == 0
        (i % 2 == 1) -> j % 2 == 1
        else -> false
    }
}